lib/chef/resource/windows_pagefile.rb

```diff
@@ -137,21 +152,27 @@ def exists?(pagefile)
         # @param [String] max the minimum size of the pagefile
         # @return [Boolean]
         def max_and_min_set?(pagefile, min, max)
-          @max_and_min_set ||= begin
-            logger.trace("Checking if #{pagefile} min: #{min} and max #{max} are set")
-            cmd = shell_out("wmic.exe pagefileset where SettingID=\"#{get_setting_id(pagefile)}\" list /format:list", returns: [0])
-            cmd.stderr.empty? && (cmd.stdout =~ /InitialSize=#{min}/i) && (cmd.stdout =~ /MaximumSize=#{max}/i)
-          end
+          logger.trace("Checking if #{pagefile} has max and initial disk size values set")
+          cmd =  "$page_file = '#{pagefile}';"
+          cmd << "$driveLetter = $page_file.split(':')[0];"
+          cmd << "$page_file_settings = Get-CimInstance -ClassName Win32_PageFileSetting -Filter \"SettingID='pagefile.sys @ $($driveLetter):'\" -Property * -ErrorAction Stop;"
```

Comment:
what happens if `pagefile` is not `[DRIVE_LETTER]:\pagefile.sys`. For example what is the resource sets it to `e:\blah\file.sys`

Reply:
from all the examples I read about, the file is consistently named "pagefile.sys" and consistently located in the root of whatever drive you choose. I set the examples and the code to follow that pattern. Do you have an example of it being otherwise?